lib/array_equals.rb

```diff
@@ -1,5 +1,22 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  counter = 0
+
+  if array1 == nil && array2 == nil || array1 == [] && array2 == []
+    return true
+  else
+    return false
+  end
+
```

Comment:
If one is nil or the other is nil, but not both.

```suggestion
if array1.nil?  || array2.nil?
```